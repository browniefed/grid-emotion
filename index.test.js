/* eslint-env jest */
import React from 'react'
import { sheet } from 'emotion'
import serializer from 'jest-glamor-react'
import { create as render } from 'react-test-renderer'
import { Box, Flex, div } from './dist/index.cjs.js'

expect.addSnapshotSerializer(serializer(sheet))

// Box
test('Box renders', () => {
  const json = render(<Box m={2} px={3} />).toJSON()
  expect(json).toMatchSnapshot()
})

test('Box renders with props', () => {
  const json = render(<Box m={[1, 2]} px={[1, 2]} w={1} flex="1 1 auto" />)
  expect(json).toMatchSnapshot()
})

// Flex
test('Flex renders', () => {
  const flex = render(<Flex />)
  expect(flex).toMatchSnapshot()
})

test('Flex renders with props', () => {
  const flex = render(
    <Flex wrap direction="column" align="center" justify="space-between" />
  )
  expect(flex).toMatchSnapshot()
})

test('Flex renders with column prop', () => {
  const flex = render(<Flex column />)
  expect(flex).toMatchSnapshot()
})

test('Flex renders with responsive props', () => {
  const flex = render(
    <Flex
      wrap={[true, false]}
      direction={['column', 'row']}
      align={['stretch', 'center']}
      justify={['space-between', 'center']}
    />
  )
  expect(flex).toMatchSnapshot()
})

// div
test('div removes grid-styled props', () => {
  const json = render(
    React.createElement(div, {
      id: 'hi',
      className: 'beep',
      width: 0.5,
      color: 'blue',
      fontSize: 4,
      wrap: true
    })
  ).toJSON()
  expect(json.props.id).toEqual('hi')
  expect(json.props.className).toEqual('beep')
  expect(json.props.width).toEqual(undefined)
  expect(json.props.color).toEqual(undefined)
  expect(json.props.fontSize).toEqual(undefined)
  expect(json.props.wrap).toEqual(undefined)
})

test('div accepts an is prop to change elements', () => {
  const json = render(
    React.createElement(div, {
      is: 'h2'
    })
  ).toJSON()
  expect(json.type).toBe('h2')
})
