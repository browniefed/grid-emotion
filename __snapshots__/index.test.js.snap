// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders 1`] = `
.glamor-0 {
  box-sizing: border-box;
  margin: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

<div
  className="glamor-0"
/>
`;

exports[`Box renders with props 1`] = `
.glamor-0 {
  box-sizing: border-box;
  width: 100%;
  margin: 4px;
  padding-left: 4px;
  padding-right: 4px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

@media screen and (min-width:40em) {
  .glamor-0 {
    margin: 8px;
    padding-left: 8px;
    padding-right: 8px;
  }
}

<div
  className="glamor-0"
/>
`;

exports[`Flex renders 1`] = `
.glamor-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="glamor-0"
/>
`;

exports[`Flex renders with column prop 1`] = `
.glamor-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="glamor-0"
  column={true}
/>
`;

exports[`Flex renders with props 1`] = `
.glamor-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
  className="glamor-0"
  direction="column"
/>
`;

exports[`Flex renders with responsive props 1`] = `
.glamor-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media screen and (min-width:40em) {
  .glamor-0 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media screen and (min-width:40em) {
  .glamor-0 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<div
  className="glamor-0"
  direction={
    Array [
      "column",
      "row",
    ]
  }
/>
`;
